src/redux/rocketsSlice.js

```diff
@@ -0,0 +1,73 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  rockets: [],
+  isLoading: false,
+  error: null,
+};
+
+export const getRockets = createAsyncThunk('rockets/getRockets', async () => {
+  try {
+    const response = await fetch('https://api.spacexdata.com/v3/rockets');
+    if (!response.ok) {
+      throw new Error('Failed to fetch rockets');
+    }
+
+    const data = await response.json();
+    return data.map((rocket) => ({
+      id: rocket.id,
+      rocket_name: rocket.rocket_name,
+      description: rocket.description,
+      flickr_images: rocket.flickr_images,
+      isReserved: false,
+    }));
+  } catch (error) {
+    throw new Error('Failed to fetch rockets');
+  }
+});
+
+const rocketsSlice = createSlice({
+  name: 'rockets',
+  initialState,
+  reducers: {
+    bookingRocket: (state, action) => {
+      const id = parseInt(action.payload, 10);
+      state.rockets = state.rockets.map((rocket) => {
+        if (rocket.id === id) {
+          // console.log(`Rocket ${rocket.id} is reserved`);
+          return { ...rocket, isReserved: true };
+        }
+        return rocket;
+      });
+    },
+    cancelBooking: (state, action) => {
+      const id = parseInt(action.payload, 10);
+      state.rockets = state.rockets.map((rocket) => {
+        if (rocket.id === id) {
+          // console.log(`Rocket ${rocket.id} reserve was canceled`);
```

Comment:
Kindly remove all commented lines from your code to enable you have a clean code that is easily maintainable 😄

Reply:
Solved